metering/reports/metering-writing-custom-report-queries.adoc

```diff
@@ -0,0 +1,396 @@
+
+toc::[]
+
+Metering allows you to add and expand upon build-in reports and metrics by using CustomResources to gather new metrics and create new custom queries and reports. The primary CustomResources that allow this are the ReportDataSource and the ReportQuery.
+
+A ReportDataSource can configure metering to execute additional custom Prometheus queries and store the metrics collected.
+Once the data has been collected a ReportQuery can be used to analyze the metrics.
+
+This guide walks through collecting new Prometheus metrics, then writing a custom SQL query to analyze these metrics. Review the documentation on ReportDataSources and ReportDataQueries before you read this guide, or as needed while you read through.
+
+== Setup
+
+This guide assumes you have already installed metering and have Prometheus running in your cluster. 
+
+In many of the examples below you run a Prometheus query outside of the metering operator. To to this we will access Prometheus directly using the default exposed route. Run the following command to verify the route for Prometheus:
+
+----
+$ oc get route -n openshift-monitoring --field-selector metadata.name==prometheus-k8s
+
+NAME                HOST/PORT                                                                                           PATH   SERVICES            PORT    TERMINATION          WILDCARD
+alertmanager-main   alertmanager-main-openshift-monitoring.apps.ci-ln-qdhgyvb-d5d6b.origin-ci-int-aws.dev.rhcloud.com          alertmanager-main   web     reencrypt/Redirect   None
+grafana             grafana-openshift-monitoring.apps.ci-ln-qdhgyvb-d5d6b.origin-ci-int-aws.dev.rhcloud.com                    grafana             https   reencrypt/Redirect   None
+prometheus-k8s      prometheus-k8s-openshift-monitoring.apps.ci-ln-qdhgyvb-d5d6b.origin-ci-int-aws.dev.rhcloud.com             prometheus-k8s      web     reencrypt/Redirect   None
+----
+
+Next, open up your web browser to the `prometheus-k8s` URL listed under `HOST/PORT` column in the output from the command above. If you have trouble at this point then you may need to find the documentation for your Prometheus installaion on how to access the Prometheus Web UI.
+
+For the purposes of this guide, we are going to report on the `kube_deployment_status_replicas_unavailable` metric that is produced by `kube-state-metrics`, which should be available in your Prometheus instance if you followed a default {product-title} installation.
+
+To validate this, open your web browser to your Prometheus UI and in the box above the `Execute` button, enter in `kube_deployment_status_replicas_unavailable`. Then click the `Execute` button.
+A list of metrics shows up. If you have an empty list, it is possible that you do not have `kube-state-metrics` running, or there is a configuration issue with Prometheus.
+
+== Collecting additional Prometheus metrics
+
+The first thing to figure out is what question do we want to ask, and what information is needed to ask it.
```

Comment:
This sounds like a tutorial rather than a technical document. 
Maybe< _When creating additional metrics, determine what data you need and what information is needed to acquire that data._ Or something like that.